core/src/main/kotlin/fr/sncf/osrd/api/pathfinding/PathPropUtils.kt

```diff
@@ -0,0 +1,142 @@
+package fr.sncf.osrd.api.pathfinding
+
+import com.google.common.collect.Iterables
+import fr.sncf.osrd.graph.Pathfinding.EdgeRange
+import fr.sncf.osrd.railjson.schema.common.graph.EdgeDirection
+import fr.sncf.osrd.railjson.schema.infra.trackranges.RJSDirectionalTrackRange
+import fr.sncf.osrd.railjson.schema.schedule.RJSTrainPath
+import fr.sncf.osrd.sim_infra.api.*
+import fr.sncf.osrd.sim_infra.impl.ChunkPath
+import fr.sncf.osrd.sim_infra.impl.buildChunkPath
+import fr.sncf.osrd.utils.Direction
+import fr.sncf.osrd.utils.indexing.DirStaticIdx
+import fr.sncf.osrd.utils.indexing.MutableDirStaticIdxArrayList
+import fr.sncf.osrd.utils.units.Distance.Companion.fromMeters
+import fr.sncf.osrd.utils.units.Length
+import fr.sncf.osrd.utils.units.Offset
+import fr.sncf.osrd.utils.units.meters
+
+object PathPropUtils {
```

Comment:
Might be worth to look into moving some of these into `PathProperties`

Reply:
Actually, there's a reason for this separation: moving it to `PathProperties` would add dependencies to `osrd.main` (which isn't something we'd want to do)